ee/app/ldap-enterprise/server/ldapEnterprise.js

```diff
@@ -43,9 +51,23 @@ export const getLdapTeamsByUsername = (username, ldap) => {
 		scope: ldap.options.User_Search_Scope || 'sub',
 		sizeLimit: ldap.options.Search_Size_Limit,
 	};
-	const getLdapTeams = (ldapUserGroups) => ldapUserGroups.filter((field) => field && field.ou).map((field) => field.ou);
 	const ldapUserGroups = ldap.searchAllSync(ldap.options.BaseDN, searchOptions);
-	return Array.isArray(ldapUserGroups) ? getLdapTeams(ldapUserGroups) : [];
+
+	if (!Array.isArray(ldapUserGroups)) {
+		return [];
+	}
+
+	const items = ldapUserGroups.filter((field) => field && field.ou).map((field) => (typeof field.ou === 'string' ? field.ou.split(',') : field.ou));
+	const groups = [];
+	for (const item of items) {
```

Comment:
this `for` is doing exactly what the [reduce below](https://github.com/RocketChat/Rocket.Chat/pull/21768/files#diff-8708b264c4665700932d0e8b2a5f5a665e54084357fadfb5f8cb957e998ca380R105) is doing right? just flattening an array.. maybe you could use `reduce` in both places?

Reply:
oh, just realized you used in other places too.. maybe could be even better creating a reducer to use on all those places =)

Reply:
I think the Node version we use supports the `[].flat` function 👀 isn't it?

Reply:
wut?! I didn't know that 🤔 `[].flat` is better indeed

Reply:
`[].flat` is like the `flattenDeep` method of lodash, It's pretty useful 🎉